charts/ingress-nginx/templates/controller-servicemonitor.yaml

```diff
@@ -11,8 +11,11 @@ metadata:
   labels:
     {{- include "ingress-nginx.labels" . | nindent 4 }}
     app.kubernetes.io/component: controller
-  {{- if .Values.controller.metrics.serviceMonitor.additionalLabels }}
-    {{- toYaml .Values.controller.metrics.serviceMonitor.additionalLabels | nindent 4 }}
+  {{- with .Values.controller.metrics.serviceMonitor.additionalLabels }}
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+  {{- with .Values.controller.metrics.serviceMonitor.annotations }}
+  annotations: {{- toYaml . | nindent 4 }}
```

Comment:
```suggestion
  {{- if .Values.controller.metrics.serviceMonitor.additionalLabels }}
    {{- toYaml .Values.controller.metrics.serviceMonitor.additionalLabels | nindent 4 }}
  {{- end }}
  {{- if .Values.controller.metrics.serviceMonitor.annotations }}
  annotations: {{ toYaml .Values.controller.metrics.serviceMonitor.annotations | nindent 4 }}
```

Reply:
I know `with` is tempting in this case, I did the same for some time, but I'd not use it for simple strings for two reasons (anymore):

1. It does not bring any real advantage over a simple `if` as you're still using the same value in the end.
2. It makes code harder to read and might bring unintended behavior as it changes to context on the inside, which is sometimes already hard to handle.

So I'd ask you to stick to the simple `if` here please. Thanks!

Reply:
Thanks for the suggestion @Gacko. Applied!